.github/shared/src/doc-preview.js

```diff
@@ -0,0 +1,138 @@
+// @ts-check
+const DOCS_NAMESPACE = "_swagger_specs";
+const SPEC_FILE_REGEX =
+  "(specification/)+(.*)/(resourcemanager|resource-manager|dataplane|data-plane|control-plane)/(.*)/(preview|stable|privatepreview)/(.*?)/(example)?(.*)";
+
+/**
+ * Extract swagger file metadata from path.
+ * @param {string} specPath
+ * @returns {{
+ *   path: string,
+ *   serviceName: string,
+ *   serviceType: string,
+ *   resourceProvider: string,
+ *   releaseState: string,
+ *   apiVersion: string,
+ *   fileName: string
+ * } | null}
+ */
+export function parseSwaggerFilePath(specPath) {
+  const m = specPath.match(SPEC_FILE_REGEX);
+  if (!m) {
+    console.log(`Path "${specPath}" does not match expected swagger file pattern.`);
```

Comment:
is this an exception and we should throw, or do we just want to ignore paths not matching regex?

Reply:
Previous code only returns null. I think an exception is better here as it would be useful to know if we encounter this issue.